frontends/concrete-python/concrete/fhe/representation/graph.py

```diff
@@ -194,6 +197,172 @@ def evaluate(
+
+            result += f"[%{index}: {details}]"
+
+            return result
+
+        graph = self.graph
+        output_nodes = set(self.output_nodes.values())
+
+        attributes = {
+            node: {
+                "label": get_label(i, node),
+                "color": get_color(node),
+                "penwidth": 2,  # double thickness for circles
+                "peripheries": 2 if node in output_nodes else 1,  # two circles for output nodes
+            }
+            for i, node in enumerate(nx.lexicographical_topological_sort(graph))
+        }
+        nx.set_node_attributes(graph, attributes)
+
+        for edge in graph.edges(keys=True):
+            idx = graph.edges[edge]["input_idx"]
+            graph.edges[edge]["label"] = f" {idx} "
+
+        try:
+            agraph = nx.nx_agraph.to_agraph(graph)
+        except ImportError as error:  # pragma: no cover
+            if "pygraphviz" in str(error):
+                message = (
+                    "Graph.draw requires `full` version of Concrete. "
+                    "See https://docs.zama.ai/concrete/getting-started/installing."
+                )
+                raise ImportError(message) from error
+
+            raise
+
+        agraph.graph_attr["rankdir"] = "LR" if horizontal else "TB"
+        agraph.layout("dot")
+
+        if save_to is None:
+            with tempfile.NamedTemporaryFile(suffix=".png", delete=False) as tmp:
+                # we need to change the permissions of the temporary file
+                # so that it can be read by all users
+
+                # (https://stackoverflow.com/a/44130605)
+
+                # get the old umask and replace it with 0o666
+                old_umask = os.umask(0o666)
+
+                # restore the old umask back
+                os.umask(old_umask)
+
+                # combine the old umask with the wanted permissions
+                permissions = 0o666 & ~old_umask
+
+                # set new permissions
+                os.chmod(tmp.name, permissions)
+
+                save_to_str = str(tmp.name)
+        else:
+            save_to_str = str(save_to)
+
+        agraph.draw(save_to_str)
+
+        if show:  # pragma: no cover
+            try:
+                # pylint: disable=import-outside-toplevel
+                import matplotlib.pyplot as plt
+                from PIL import Image
+
+                # pylint: enable=import-outside-toplevel
+            except ImportError as error:
+                if "matplotlib" in str(error) or "pillow" in str(error) or "PIL" in str(error):
+                    message = (
+                        "Graph.draw with show=True requires `full` version of Concrete. "
+                        "See https://docs.zama.ai/concrete/getting-started/installing."
+                    )
+                    raise ImportError(message) from error
+
+                raise
+
+            plt.close("all")
+            plt.figure()
```

Comment:
could put a tile here, imagine you want to use the feature to show in a presentation.
could be a parameter or the name of the circuit

Reply:
I think if they want it, they can explicitly use matplotlib for that because where would we stop? Like they might want to edit the figure size, padding around the figure etc. It's not that hard either:
```python
import matplotlib.pyplot as plt
from PIL import Image

img = Image.open(circuit.draw())
plt.imshow(img)
img.close()

plt.title("Foo")
plt.axis("off")
plt.show()
```

Reply:
Yes it's always possible to hackaround, but at some point you'll need to add options on the rendering.